src/classes/Expect.cls

```diff
@@ -57,47 +57,47 @@ public class Expect {
         }
 
         public void shouldEqual(Object actual, String message) {
-            assert(expected == actual, message);
+            assert(JSON.serialize(expected).equals(JSON.serialize(actual)), message);
```

Comment:
yeah, I definitely get it!  Definitely don't have to merge this stuff - I consider it a good learning experience in testing equality in different ways